anvil_consortium_manager/models.py

```diff
@@ -72,12 +75,76 @@ def anvil_import(cls, billing_project_name):
         return billing_project
 
 
+class UserEmailEntry(TimeStampedModel, models.Model):
+    """A model to store emails that users could link to their AnVIL account after verification."""
+
+    uuid = models.UUIDField(default=uuid.uuid4)
+    """UUID for use in urls."""
+
+    email = models.EmailField()
+    """The email entered by the user."""
+
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL, null=True, on_delete=models.CASCADE
+    )
+    """The user who created the record."""
+
+    date_verification_email_sent = models.DateTimeField()
+    """Most recent date that a verification email was sent."""
+
+    can_be_verified = models.BooleanField(default=True)
+    """Indicator of whether this UserEmailEntry can be verified.
+
+    This field will be set to False when a user verifies an email for all records associated with
+    either the user or the email that they verified.
+    """
+
+    verified_account = models.ForeignKey(
```

Comment:
Should this be defined here, or give `Account` a foreign key to `UserEmailEntry`? I think I like the latter better because it follows the data flow better. You can still see get to the account from a user_email_entry object due to the Django backwards relationships.